src/hotspot/share/runtime/suspendResumeManager.hpp

```diff
@@ -59,6 +59,9 @@ class SuspendResumeManager {
 
   void set_suspended(bool to, bool register_vthread_SR);
 
+  // The specific 'set_suspended' implementation for self suspend.
+  void set_suspended_with_id(int64_t id, bool register_vthread_SR);
```

Comment:
```suggestion
  // Sets the suspended state to `to`, applying to vthreads if `register_vthread_SR` is true.
  void set_suspended(bool to, bool register_vthread_SR);

  // Sets the suspended state to true, applying to vthreads if `register_vthread_SR` is true.
  // Applied to the thread with the given `id` and used when we can't extract the thread oop safely.
  void set_suspended_with_id(int64_t id, bool register_vthread_SR);
```
These comments are far from perfect as it is actually hard to explain exactly what `thread` is being operated on - if any!

Reply:
Okay I was trying to understand exactly what these methods were doing but I'm finding it somewhat difficult to explain that. But my point is to document that the first method sets the state to true/false as directed, whilst the second always sets to true. But it gets harder to describe things when the id variant is only needed when `register_vthread_SR` is true.

Reply:
The `set_suspended` maintained _suspended flag for individual flags. While to support fast suspending of all virtual threads it uses additional mechanisms.
So, it is rather `set_suspended_status` or something like this. 
I don't know what would be good name for this logic.